File: perl/fizzbuzz.pl
fizzbuzz();

sub fizzbuzz {
    @output_values;

    for my $index (1..100) {
        $fizz_buzz_string = '';

        if ($index % 3 == 0) {
            $fizz_buzz_string .= 'Fizz';
        }
        if ($index % 5 == 0) {
            $fizz_buzz_string .= 'Buzz';
        }

        push @output_values, $fizz_buzz_string ? $fizz_buzz_string : $index;
    }

    print join(' ', @output_values) . "\n";
}

File: ruby/fizzbuzz.rb
def fizzbuzz()
    output_values = []

    for i in 1..100
        fizz_buzz_string = ''

        if i % 3 == 0
            fizz_buzz_string += 'Fizz'
        end
        if i % 5 == 0
            fizz_buzz_string += 'Buzz'
        end

        output_values.push(fizz_buzz_string.length > 0 ? fizz_buzz_string : i)
    end

    puts output_values.join(' ')
end

fizzbuzz()
